ambari/src/main/java/io/brooklyn/ambari/rest/RequestCheckRunnable.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.brooklyn.ambari.rest;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.HttpClient;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.jayway.jsonpath.JsonPath;
+
+import brooklyn.util.http.HttpTool;
+import brooklyn.util.repeat.Repeater;
+import brooklyn.util.time.Duration;
+import io.brooklyn.ambari.rest.domain.Request;
+
+public class RequestCheckRunnable implements Runnable {
+
+    private static final List<String> VALID_STATES = ImmutableList.of("IN_PROGRESS", "COMPLETED");
+
+    private final Builder builder;
+
+    public static Builder check(Request request) {
+        return new Builder(request);
+    }
+
+    protected RequestCheckRunnable(Builder builder) {
+        this.builder = builder;
+    }
+
+    @Override
+    public void run() {
+        boolean done = Repeater.create(String.format("Request %s status check", builder.request.toString()))
+                .every(Duration.ONE_SECOND)
```

Comment:
We could use exponential back-off but I'm not sure if it is the best thing to do as you might hit the timeout before the repeater retries to verify the result. Does it make sens?